File: 10_mouse_input_review_peer/program_files/python/11mouseshow.py
from splashkit import *

def main():

    spotlight_radius = 100
    window_width, window_height = 800, 600
    window = open_window("Mouse Visibility in Spotlight", window_width, window_height)

    spotlight_center = point_at(window_width / 2, window_height / 2)

    mouse_visible = True

    while not window_close_requested(window):
        # Process user input events
        process_events()

        # Get the current mouse position
        mouse_pos = mouse_position()

        # Calculate distance from the mouse to the spotlight center
        dx = mouse_pos.x - spotlight_center.x
        dy = mouse_pos.y - spotlight_center.y
        distance = (dx ** 2 + dy ** 2) ** 0.5

        # Determine whether to show or hide the mouse
        if distance <= spotlight_radius:
            if not mouse_visible:
                show_mouse()  # Show the mouse
                mouse_visible = True
        else:
            if mouse_visible:
                hide_mouse()  # Hide the mouse
                mouse_visible = False

        # Clear the screen with black color
        clear_screen(color_black())

        # Draw the spotlight as a circle
        fill_circle(color_white(), spotlight_center.x, spotlight_center.y, spotlight_radius)

        refresh_window(window)

    close_all_windows()

if __name__ == "__main__":
    main()

File: peer-reviews/10_mouse_input_review_peer/program_files/cpp/11mouseshow.cpp
#include "splashkit.h"

int main()
{
    // create cirle variable for emulating a "stage spot light"
    circle spotLight;

    open_window("Mouse Visibility", 800, 600);

    // create a "spot light" circle in centre of screen
    spotLight = circle_at(screen_center(), 100);

    while (!window_close_requested("Mouse Visibility"))
    {
        // process user input events
        process_events();

        // mouse disappears when not in the "spot light"
        show_mouse(point_in_circle(mouse_position(), spotLight));

        // display "spot light" on screen
        clear_screen(COLOR_BLACK);
        fill_circle(COLOR_ALICE_BLUE, spotLight);
        refresh_screen(60);
    }

    close_all_windows();

    return 0;
}